content/en/docs/architecture/step-registry.md

```diff
@@ -559,11 +572,176 @@ tests:
     test:
     - as: test
       leases:
-      # overriden by the parent section
+      # overridden by the parent section
       - resource_type: aws-quota-slice
         env: AWS_LEASED_RESOURCE
       # added to the parent section
       - resource_type: gcp-quota-slice
         env: GCP_LEASED_RESOURCE
       # …
 {{< / highlight >}}
+
+## Hierarchical Propagation
+
+Some fields of individual steps can be changed by the chains, workflows, and
+test definitions that include them. Those are: [parameters](#parameters),
+[dependencies](/docs/architecture/ci-operator#referring-to-images-in-tests), and
+[leases](#leases).
+
+Values set in parent elements will be propagated down the hierarchy. That is: a
+variable in the `env` section of a chain will propagate to all of its sub-chains
+and sub-steps, a variable in the `env` section of a workflow or test will
+propagate to all of its stages. The same applies for dependencies and leases.
+
+{{< alert title="Warning" color="warning" >}}
+As described in their individual sections, parameters and dependencies must be
+declared in all steps that use them: setting values in parent components is not
+sufficient. Basic compliance with this rule is enforced and simple cases of
+unused values will result in validation errors, but not all can be detected,
+resulting in parameter values not being set.
+{{< /alert >}}
+
+One special rule applies to list and mapping fields that are specified both in a
+test and its workflow.  Instead of completely overriding the workflow value, as
+is the case for scalar values, the two sections are merged according to the
+following rules:
+
+- Parameters and dependencies declared in the test override those in the
+  workflow if they target the same environment variable. Otherwise, the
+  resulting parameter list is the combination of both sections.
+- Leases declared in the test must not target an environment variable already
+  present in the workflow. Otherwise, the resulting lease list is the
+  combination of both sections.
+
+### Examples
+
+A step declaring a parameter:
```

Comment:
I think giving the individual examples short names and using L4 headers (####) would help to break this section down visually (and individual examples would be linkable which may be useful)